patches/api/0329-Expose-Player-Idle-Time.patch

```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: thamid-gamer <60953955+thamid-gamer@users.noreply.github.com>
+Date: Sat, 31 Jul 2021 03:42:06 -0400
+Subject: [PATCH] Expose Player Idle Time
+
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 3ec1be36b90dbedb8631135555da4b69110e4791..bba769a8d56eff5c1cf7d49c381b9a68732553fd 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -2080,6 +2080,16 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+      */
+     void resetCooldown();
+ 
++    /**
++     * Gets the time the player has been idle in milliseconds.
++     * Many types of packets sent from the player will reset the idle time.
++     * This is determined the same way the server determines the idle time
++     * to kick players based on the player-idle-timeout setting in server.properties.
```

Comment:
I'd maybe more ref that this is an impl decision rather than hard-ratifying on what it's set by, API and all